### PR TITLE
MYR-17 : Improvement for const correctness to my_hash used by myrocks

### PR DESCRIPTION
**Pull request by george-lorch**
- Copied in upstream improvement to my_hash for myrocks

### CODE REVIEW

**Comment by george-lorch** [comment]
https://jira.percona.com/browse/MYR-17
https://blueprints.launchpad.net/percona-server/+spec/myrocks-const-correct-5.6
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1459/

File: mysys/hash.c [hunk 1]
```diff
@@ -717,6 +717,14 @@ uchar *my_hash_element(HASH *hash, ulong idx)
 }
 
 
+const uchar *my_hash_const_element(const HASH *hash, ulong idx)
+{
+  if (idx < hash->records)
+    return dynamic_element(&hash->array,idx,const HASH_LINK*)->data;
```
Comment (laurynas-biveinis) [inline mysys/hash.c]: spaces after commas

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comment for bug vs blueprint as for MYR-22 (also no change required for that)

File: mysys/hash.c (hunk 1 above)
Comment (george-lorch) [reply mysys/hash.c]: Done and fixed existing issue just above as well.

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**